lib/helpers/hearing_renderer.rb

```diff
@@ -0,0 +1,358 @@
+      puts render(veteran_with_legacy_hearing, show_pii: true)
+    end
+
+    def test_breadcrumbs
+      ama_virtual_hearing = VirtualHearing.not_cancelled.where(hearing_type: Hearing.name).last
+      puts render(ama_virtual_hearing.hearing)
+    end
+
+    def render(obj, show_pii: false)
+      renderer = HearingRenderer.new(show_pii: show_pii)
+      TTY::Tree.new(renderer.structure(obj, include_breadcrumbs: true)).render
+    end
+
+    def renderable_classes
+      RENDERABLE_CLASSNAMES.map(&:constantize)
+    end
+
+    def prefix(obj)
+      klass = renderable_classes.find { |k| obj.is_a?(k) }
+      klass&.name&.underscore
+    end
+  end
+
+  attr_reader :show_pii
+
+  def initialize(show_pii: false)
+    @show_pii = show_pii
+  end
+
+  def structure(obj, include_breadcrumbs: false)
+    return if obj.nil?
+
+    children = try("#{self.class.prefix(obj)}_children", obj)&.compact || []
+    if include_breadcrumbs
+      context = calculate_breadcrumbs(obj)
+      children << { "breadcrumbs:": context } if context.present?
+    end
+    { "#{label(obj)}": children }
+  end
+
+  def label(obj)
+    return "nil" if obj.nil?
+
+    result = "#{obj.class.name} #{obj.id}"
+    details = try("#{self.class.prefix(obj)}_details", obj)&.compact
+    result += " (#{details.join(', ')})" if details.present?
+    result
+  end
+
+  def calculate_breadcrumbs(obj)
+    context = try("#{self.class.prefix(obj)}_context", obj)
+    return [] if context.blank?
+
+    [label(context), calculate_breadcrumbs(context)].flatten
+  end
+
+  def veteran_details(vet)
+    details = []
+    details += [vet.name, "FN: #{vet.file_number}"] if show_pii
+    details << "PID: #{vet.participant_id}"
+    details
+  end
+
+  def readable_date(date, include_time = true)
+    if include_time
+      date.strftime("%m-%d-%Y %I:%M%p %Z")
+    else
+      date.strftime("%m-%d-%Y")
+    end
+  end
+
+  def veteran_children(vet)
+    appeals = Appeal.where(veteran_file_number: vet.file_number)
+    appeals.sort_by { |appeal| appeal.receipt_date || Time.zone.today }
+    legacy_appeals = LegacyAppeal.fetch_appeals_by_file_number(vet.file_number)
+    legacy_appeals.sort_by { |la| la.created_at || Time.zone.today }
+
+    children = []
+    children += legacy_appeals.map { |legacy_appeal| structure(legacy_appeal) }
+    children += appeals.map { |appeal| structure(appeal) }
+    children
+  end
+
+  def get_appeal_type_conversions(appeal)
+    unformatted_versions = appeal.versions
+    versions = unformatted_versions.map do |v|
+      change = v.changeset["changed_hearing_request_type"] ||
+               v.changeset["changed_request_type"]
+      {
+        "from_type" => print_nil(Hearing::HEARING_TYPES[change[0]&.to_sym]),
+        "to_type" => Hearing::HEARING_TYPES[change[1]&.to_sym],
+        "converted_by" => User.find(v.whodunnit).css_id,
+        "converted_at" => readable_date(v.created_at)
+      }
+    end
```

Comment:
suggesting codeclimate fixes
```suggestion
    versions = unformatted_versions.map do |version|
      change = version.changeset["changed_hearing_request_type"] ||
               version.changeset["changed_request_type"]
      {
        "from_type" => print_nil(Hearing::HEARING_TYPES[change[0]&.to_sym]),
        "to_type" => Hearing::HEARING_TYPES[change[1]&.to_sym],
        "converted_by" => User.find(version.whodunnit).css_id,
        "converted_at" => readable_date(version.created_at)
      }
    end
```